### ISSUE TITLE
need testers for Switch port on real hardware

### ISSUE THREAD

**Issue by headshot2017** [headshot2017/re-plants-vs-zombies#1]
i do not own a Switch, so my tests are limited to emulators. if you have a modded Switch, i'd appreciate it if you could volunteer

**Comment by spaghetto207** [comment]
i'd be glad to test, i have a modded switch.
i am VERY shocked to see this. this is crazy!

**Comment by headshot2017** [comment]
great! we can talk here just fine. you can start by getting the NRO rom from the [Actions](https://github.com/headshot2017/re-plants-vs-zombies/actions/workflows/switch.yml) page, just click on the first workflow run in the list and scroll down to Artifacts

**Comment by spaghetto207** [comment]
Alright, i initially compiled it myself though, ill download the artifact and tell you the details once i get it set up!
Any reason to make a port to the 3DS btw? I know the DS & DSI versions exist but they are heavily downgraded to work with the system.

**Comment by headshot2017** [comment]
there's actually a 3DS port in progress but the game and engine need some heavy modifications to consume less memory (VRAM is also a concern). it currently crashes at the title screen on New 3DS, and Old 3DS doesn't get that far

![image](https://github.com/user-attachments/assets/2d5e1bae-a10e-4e74-82d1-6c02393ab95c)

**Comment by spaghetto207** [comment]
Alright, it took a bit of setup though; I initially used an extracted main.pak from steam GOTY. 
using an extracted main.pak made the loading super slow; it goes to the end of the bar before it allows you to continue, and it continues loading until it crashes from trying to compile one of the particles or reanims,
After using the properties folder and main.pak from goty, it loaded just fine but has small stutters between animations, such as when pressing start adventure, it pauses for a brief moment before showing the hand, then it takes a bit to load the initial cut-scene before crashing with no error. 
If possible, it would be nice to have an output log because it sometimes crashes with no error.
I can try to record footage from my switch, but I am away right from it right now.

**Comment by headshot2017** [comment]
oh yeah, about the compiled definitions, i have here all the reanims and particles compiled for Switch - if i ever get to release this port, it will include this data to avoid any issues on that regard
your crash during the intro cutscene might be related to that since i also experienced it on an emulator
[compiled.zip](https://github.com/user-attachments/files/16912205/compiled.zip)

**Comment by spaghetto207** [comment]
It's alright, i just got back home to test and yes, it seems to crash on the cutscene still. 
I'll make a fork and see if i can try to fix it.

**Comment by headshot2017** [comment]
hold up, i crashed on the intro cutscene after "PopCap Games Presents:"
![image](https://github.com/user-attachments/assets/ce144858-e602-4311-a353-6559578f4239)
i might have an idea on what the cause is

**Comment by headshot2017** [comment]
turns out the PC version also crashes on that cutscene, and i managed to find out why with gdb:
![image](https://github.com/user-attachments/assets/a209f9c0-eb1b-4973-8d70-1345b40ec29f)
on CutScene.cpp, the game tries to draw the PVZ logo, but the title screen assets were unloaded after exiting the titlescreen widget, and the logo was one of those assets
working on a fix for that now

**Comment by spaghetto207** [comment]
Amazing! I'll test this soon.
Have you also seen minerscale's fork? iirc it includes alot of fixes for things like this and includes vulkan support.

**Comment by headshot2017** [comment]
yes, although i haven't really used it - it's a fork of the sexy-vk branch from the original repository, but i didn't use it due to certain reasons - vulkan is unavailable on switch homebrew, and last i tried the sexy-vk branch it consumed a lot of memory which didn't really sit well with me, so i went with forking the main branch

**Comment by spaghetto207** [comment]
Alright, all is going well so far.
The only "major" issue ive seen is the seed packet pickup at the end of a level, The first time it didnt show the plant, and crashes after picking it up. I cant seem to reproduce this error again though.
Another issue are some issues ive seen in other decompiles, such as the tutorial arrows being positioned wrong, im unsure if theres more of these small issues but ill try to patch them on my fork.

**Comment by headshot2017** [comment]
> The only "major" issue ive seen is the seed packet pickup at the end of a level, The first time it didnt show the plant, and crashes after picking it up. I cant seem to reproduce this error again though.
do you remember which level that was?

**Comment by spaghetto207** [comment]
it was the first level, when you get sunflower.
it had no plant icon, just the seedpacket image and sun cost. i clicked on it only for the sun cost to show up, then crashed after a bit. i cannot seem to reproduce it

also, sometimes theres a sound glitch, its a weird beep that progressivley gets louder, this may just be my switch since its old but its never done that before. ill try to get a video of it
